comp/core/settings/component_mock.go

```diff
@@ -7,7 +7,21 @@
 
 package settings
 
+import (
+	"go.uber.org/fx"
+)
+
 // Mock implements mock-specific methods.
 type Mock interface {
 	Component
 }
+
+type MockComponent struct {
+	Mock
+}
+
+func MockModule() fx.Option {
+	return fx.Provide(func() Component {
+		return new(MockComponent)
```

Comment:
`new(MockComponent)` creates the struct `MockComponent` which contains the uninitialised embedded interface `Mock`. Any attempts to call a method from `Mock` leads to `panic: runtime error:`.